pkgs/misc/uboot/default.nix

```
@@ -543,6 +545,21 @@ in {
     filesToInstall = ["u-boot-sunxi-with-spl.bin"];
   };
 
+  # Refer to https://nixos.wiki/wiki/NixOS_on_ARM/OLIMEX_Teres-A64
+  ubootTeresA64 = buildUBoot {
+    defconfig = "teres_i_defconfig";
+    extraMeta = {
+      platforms = ["aarch64-linux"];
+      maintainers = with lib.maintainers; [ kreyren ];
+    };
+    BL31 = "${armTrustedFirmwareAllwinner}/bl31.bin";
+    # U-Boot requires crust built using OpenRISC 1000 compiler for power management
+    SCP = if stdenv.hostPlatform.isOr1k
+      then "${crustTeresA64}/scp.bin"
+      else "${pkgsCross.or1k.crustTeresA64}/scp.bin";
```

```suggestion
    SCP = "${pkgsCross.or1k.crustTeresA64}/scp.bin";
```

The situation where this will not be `pkgsCross.or1k` *will never happen*. Plainly and simply.